app/views/fields/subjects_select_field/_form.html.erb

```diff
@@ -1,11 +1,13 @@
 <%
-  field.resource.subjects = "{\"subject\"=>{\"subject_id\"=>nil, \"subject_title\"=>nil}}" if field.resource.subjects.nil?
   subject_list = eval field.resource.subjects
```

Comment:
Is it possible to avoid this `eval`?

Reply:
I haven't found a way, but am open to alternatives.

Reply:
Where is field.resource.subjects getting serialized ?  it should be possible to serialize that either JSON or YAML which are data only vs code... eval is frowned upon because it introduces big security holes.

Reply:
Are these still correct? 
https://github.com/tulibraries/Press-6/blob/main/spec/factories/book.rb#L13-L14
I am a little confused by the hash vs array format...maybe we need to add a factory for each possible type of value fo the subjects?

Reply:
In this PR I wrote the test against that factory and removed all the hash parsing logic:
https://github.com/tulibraries/Press-6/pull/57
If there are other possible permutations of `book.subject` I can add a test for each.

Reply:
> Where is field.resource.subjects getting serialized ? it should be possible to serialize that either JSON or YAML which are data only vs code... eval is frowned upon because it introduces big security holes.

https://github.com/tulibraries/Press-6/blob/35fef89a6afc80acad0e2cd649a0e7b1d35357b5/app/services/sync_service/books.rb#L74

example that creates an array:

<subjects>
<subject><subject_id>1032</subject_id><subject_title>American Studies</subject_title></subject>
<subject><subject_id>1064</subject_id><subject_title>Latin American/Caribbean Studies</subject_title></subject>
<subject><subject_id>1069</subject_id><subject_title>Music and Dance</subject_title></subject></subjects>

example that created a hash:

<subjects>
<subject><subject_id>1032</subject_id><subject_title>American Studies</subject_title></subject></subjects>

example that creates a nil which needs to be converted to a hash:

<subjects>
<subject><subject_id></subject_id><subject_title></subject_title></subject></subjects>

Reply:
> Are these still correct?
> https://github.com/tulibraries/Press-6/blob/main/spec/factories/book.rb#L13-L14
> I am a little confused by the hash vs array format...maybe we need to add a factory for each possible type of value fo the subjects?

That is pseudo factory code and needs to be fleshed out as the functionality has now been implemented. Specifically we can test that the subjects array/hash will create a select list with those array/hash elements as options, if that test is desired.